app/src/main/java/com/alexbar/layoutbasic/screen/confirm_payment/screens/ConfirmPaymentScreen.kt

```diff
@@ -0,0 +1,161 @@
+import com.alexbar.layoutbasic.utils.ConfirmPayment.total_amount
+import com.alexbar.layoutbasic.utils.ConfirmPayment.welcome_message
+import kotlinx.coroutines.launch
+
+@SuppressLint("MutableCollectionMutableState")
+@Composable
+fun ConfirmPaymentScreen() {
+    val scope = rememberCoroutineScope()
+    val snackBarHostState = remember { SnackbarHostState() }
+    var selectedIndex by remember { mutableIntStateOf(0) }
+    var showNewCreditCardDialog by remember { mutableStateOf(false) }
+    val paymentMethods = listOf(payment_method_user_balance, payment_method_credit_card)
+    val currentUserBalance = current_user_balance
+    var creditCards by remember { mutableStateOf(mutableListOf<CreditCardInfo>()) }
+
+    LaunchedEffect(Unit) {
+        scope.launch {
+            snackBarHostState
+                .showSnackbar(welcome_message)
+        }
+    }
+
+    ConstraintLayout(modifier = Modifier
+        .fillMaxSize()
+        .background(Color.White)
+    ) {
+        val (header, paymentMethodSegmentedControl, payButton, snackBar,
+            userBalance, emptyCreditCards, listCreditCards) = createRefs()
+
+        HeaderConfirmPayment(
+            userBalance = total_amount,
+            modifier = Modifier.constrainAs(header){}
+        )
+
+        SegmentedControlConfirmPayment(
+            title = payment_method_title,
+            options = paymentMethods,
+            selectedIndex = selectedIndex,
+            onOptionSelected = { index -> selectedIndex = index },
+            modifier = Modifier.constrainAs(paymentMethodSegmentedControl) {
+                top.linkTo(header.bottom)
+            }
+        )
+
+        if (selectedIndex == 0) {
+            UserBalance(
+                currentBalance = currentUserBalance,
+                modifier = Modifier.constrainAs(userBalance) {
+                    top.linkTo(paymentMethodSegmentedControl.bottom)
+                }
+            )
+        } else {
+            if (creditCards.isEmpty()) {
+                EmptyCreditCards(
+                    modifier = Modifier.constrainAs(emptyCreditCards) {
+                        top.linkTo(paymentMethodSegmentedControl.bottom)
+                    },
+                    onClickAddNewCard = { showNewCreditCardDialog = true }
+                )
+            } else {
+                CreditCardList(
+                    creditCards,
+                    modifier = Modifier.constrainAs(listCreditCards) {
+                        top.linkTo(paymentMethodSegmentedControl.bottom)
+                    }
+                )
+            }
+        }
+
+        if (showNewCreditCardDialog)
+            AddCreditCardDialog(
+                onCreateCard = {
+                    creditCards.add(it)
+                    showNewCreditCardDialog = false
+                }
+            ) { showNewCreditCardDialog = false }
+
+        Button(
+            onClick = {
+                if (selectedIndex == 1)
```

Comment:
for this block :
```

if (selectedIndex == 1) {
    scope.launch {
        val result = snackBarHostState.showSnackbar(
            message = no_credit_cards_message,
            actionLabel = add_credit_cards_button,
            duration = SnackbarDuration.Short
        )
        if (result == SnackbarResult.ActionPerformed) {
            showNewCreditCardDialog = true
        }
    }
}
```
always my suggestion is using when instead of if , but if you have single condition () as I can see :
`SnackbarResult.Dismissed -> {}`
is empty , so is it better to use if for whole the code